scripts/addon/build.py

```diff
@@ -0,0 +1,361 @@
+
+    guide_json = manifest["guide"]
+    if not "id" in guide_json:
+        exit(f"Guide {filename} does not have an id")
+    if not "title" in guide_json:
+        exit(f"Guide {filename} does not have a title")
+    if not "subtitle" in guide_json:
+        exit(f"Guide {filename} does not have a subtitle")
+    if not "blocks" in guide_json:
+        exit(f"Guide {filename} does not have a blocks")
+
+    guide_id = guide_json["id"]
+    title_id = f"guide.{guide_id}.title"
+    guide_strings[title_id] = {
+        "value": guide_json["title"],
+        "comments": guide_json.get("title_comment", "Title for a guide view"),
+    }
+    subtitle_id = f"guide.{guide_id}.subtitle"
+    guide_strings[subtitle_id] = {
+        "value": guide_json["subtitle"],
+        "comments": guide_json.get("subtitle_comment", "Subtitle for a guide view"),
+    }
+
+    for block in guide_json["blocks"]:
+        if not "id" in block:
+            exit(f"Guide {filename} does not have an id for one of the blocks")
+        if not "type" in block:
+            exit(f"Guide {filename} does not have a type for block id {block['id']}")
+        if not "content" in block:
+            exit(f"Guide {filename} does not have a content for block id {block['id']}")
+
+        block_id = block["id"]
+        block_string_id = f"guide.{guide_id}.block.{block_id}"
+        block_default_comment = comment_types.get(block["type"], "")
+        if block_string_id in guide_strings:
+            exit(f"Duplicate block enum {block_string_id} when parsing {filename}")
+
+        if not isinstance(block["content"], list):
+            guide_strings[block_string_id] = {
+                "value": block["content"],
+                "comments": block.get("comment", block_default_comment),
+            }
+            continue
+
+        for subblock in block["content"]:
+            if not "id" in subblock:
+                exit(
+                    f"Guide {filename} does not have an id for one of the subblocks of block {block_id}"
+                )
+            if not "content" in subblock:
+                exit(
+                    f"Guide file {filename} does not have a content for subblock id {subblock['id']}"
+                )
+
+            subblock_id = subblock["id"]
+            subblock_string_id = f"guide.{guide_id}.block.{block_id}.{subblock_id}"
+            if subblock_string_id in guide_strings:
+                exit(
+                    f"Duplicate sub-block enum {subblock_string_id} when parsing {filename}"
+                )
+
+            guide_strings[subblock_string_id] = {
+                "value": subblock["content"],
+                "comments": subblock.get("comment", block_default_comment),
+            }
+
+    return guide_strings
+
+
+def write_en_language(filename, strings):
+    ts = ET.Element("TS")
+    ts.set("version", "2.1")
+    ts.set("language", "en")
+
+    context = ET.SubElement(ts, "context")
+    name = ET.SubElement(context, "name")
```

Comment:
This is not used anywhere.

Reply:
Right, but the TS file seems to need an empty `<name/>`

Reply:
I assume you should be able to just call `ET.SubElement(context, "name")`, without assigning it to a variable?